notebooks/COS/ViewData/ViewData.ipynb

```diff
@@ -0,0 +1,1452 @@
+    "2. Plot the NUV spectrum from stripes NUVA and NUVB\n",
+    "3. Plot the flagged data in another color to see where the removed data is"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Your answer here\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=BinningV></a>\n",
+    "## 3.2. Binning \n",
+    "\n",
+    "Up until now, the data we've been working with has one value per pixel of the detector.\n",
+    "COS data is generally binned to a resolution element, or [resel](https://hst-docs.stsci.edu/cosihb/chapter-3-description-and-performance-of-the-cos-optics/3-2-size-of-a-resolution-element), when working with it. These resels are *generally* the following sizes *in pixels*.\n",
+    "\n",
+    "||FUV|NUV|\n",
+    "|-|---|---|\n",
+    "|**Dispersion Axis**|6|3|\n",
+    "|**Cross-Dispersion Axis**|10|3|\n",
+    "\n",
+    "So, for the FUV, we generally want to preliminarily bin our x1d or x1dsum data by 6.\n",
+    "\n",
+    "In the next few cells below, we define two functions:\n",
+    "1. `downsample_1d`, for binning an array \n",
+    "2. `binByResel`, which applies this binning to an entire COS NUV or FUV dataset\n",
+    "\n",
+    "<a id=BinfunV></a>\n",
+    "### 3.2.1. Creating Some Useful Functions for Binning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from Scripts.Useful_COS_Functions import downsample_1d"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to reading through this first function `downsample_1d`  in the `Scripts` folder, we can learn about it by calling the python function `help`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "help(downsample_1d)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### We see that the function `downsample_1d` can optionally (1) average a given bin's in quadrature and/or (2) weight the averages by a value.\n",
+    "1. Working in quadrature is useful for dealing with errors. See [Uncertainty propogration](https://en.wikipedia.org/wiki/Propagation_of_uncertainty).\n",
+    "2. Some pixels receive more total exposure time than their neighbors, because of how x1d and x1dsum files are created. Pixels with more exposure time should be weighted higher.\n",
+    "\n",
+    "##### In the next cell, we import a function, `binByResel`, that applies this downsampling selectively to all the data we have read in from a single COS x1dsum file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from Scripts.Useful_COS_Functions import binByResel\n",
```

Comment:
function names follow the same conventions - instead of `binByResel`, `bin_by_resel`